Clean up naming and comments

soulitzer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#56964 Clean up naming and comments**
* #55871 Refactor fast gradcheck

This PR does many things but does not update any logic:
 - Prefixes all function names that are not `gradcheck`, `gradgradcheck`, `get_numerical_jacobian`, and `get_analytical_jacobian` with underscore to indicate that they aren't part of the public API (https://github.com/pytorch/pytorch/issues/55714).
 - Improve naming to avoid referencing Jacobian rows or Jacobian cols when we really mean vjp and jvp as suggested by @zou3519
 - Try to reduce comment line length so they are more consistent and easier to read
 - Other misc improvements to documentaiton

Differential Revision: [D28096571](https://our.internmc.facebook.com/intern/diff/D28096571)

soulitzer committed:
Clean up naming and comments

[ghstack-poisoned]

soulitzer committed:
Update on "Clean up naming and comments"




[ghstack-poisoned]

soulitzer committed:
Update on "Clean up naming and comments"


This PR does many things but does not update any logic:
 - Prefixes all function names that are not `gradcheck`, `gradgradcheck`, `get_numerical_jacobian`, and `get_analytical_jacobian` with underscore to indicate that they aren't part of the public API (https://github.com/pytorch/pytorch/issues/55714).
 - Improve naming to avoid referencing Jacobian rows or Jacobian cols when we really mean vjp and jvp as suggested by @zou3519
 - Try to reduce comment line length so they are more consistent and easier to read
 - Other misc improvements to documentaiton


[ghstack-poisoned]

albanD commented on torch/autograd/gradcheck.py:
```diff
@@ -11,68 +11,68 @@
-    # the returned tensor will share storage with the original. Otherwise, for opaque tensor
-    # types like mkldnn, a copy is returned.
+def _iter_tensor(x_tensor):
+    # (Only used for slow gradcheck) Returns an generator that yields at each iter:
```
a generator

albanD replied:
yields the following elements at each iteration

albanD commented on torch/autograd/gradcheck.py:
```diff
@@ -11,68 +11,68 @@
+def _iter_tensor(x_tensor):
+    # (Only used for slow gradcheck) Returns an generator that yields at each iter:
+    #  1) a tensor: the same tensor is returned across all iterations. The tensor
+    #     is not the same as the original - it is prepared so that it can be
```
original -> x_tensor given as input?

albanD commented on torch/autograd/gradcheck.py:
```diff
@@ -11,68 +11,68 @@
+    #     modified in-place. Depending on whether the input tensor is strided,
+    #     sparse, or dense, the returned tensor may or may not share storage with
+    #     the original.
+    #  2) index into the tensor in dictionary order
```
What is dictionary order?
You mean a tuple of indices that can be used with advanced indexing?

soulitzer replied:
Dictionary order is lexical ordering i.e. (0, 0), (0, 1) (1, 0) (1, 1). Its the order in which `product(*[range(m) for m in x_tensor.size()])` yields the tuples of indices.

albanD commented on torch/autograd/gradcheck.py:
```diff
@@ -112,16 +109,17 @@ def get_stride(size):
-    """Computes the numerical jacobian for a given fn and inputs. Returns M * N jacobians
-    where M is the number of input tensors that require grad, and N is the number of output
-    float/complex tensors.
+    """Computes the numerical Jacobian of `fn` at the provided input and with
```
Maybe it would be clearer to say, Jacobian of `fn(inputs)` wrt to `target`. If not specified, targets are the input.

Side question: Is this target argument actually used?

soulitzer replied:
Yes, used in common_nn

albanD commented on torch/autograd/gradcheck.py:
```diff
@@ -112,16 +109,17 @@ def get_stride(size):
-    where M is the number of input tensors that require grad, and N is the number of output
-    float/complex tensors.
+    """Computes the numerical Jacobian of `fn` at the provided input and with
+    respect to `target`, if provided. Returns M * N Jacobians where M is the number
```
Interesting, the convention we usually use is M for the number of outputs and N for the number of inputs and a Jacobian of size MxN.
Is this comment actually accurate?

soulitzer replied:
Changed differentiable outputs -> non-integral outputs
input tensors -> tensors in `target`

albanD commented on torch/autograd/gradcheck.py:
```diff
@@ -206,28 +208,29 @@ def compute_numerical_jacobian_cols(jvp_fn, delta, input_is_complex) -> List[tor
-def prepped_input(input: torch.Tensor, maybe_perturbed_input: Optional[torch.Tensor],
-                  fast_mode=False) -> torch.Tensor:
-    # Prepares the inputs to be passed into the function while including the new modified input.
+def _prepped_input(input: torch.Tensor, maybe_perturbed_input: Optional[torch.Tensor],
```
nit: this one should be `_prepare_input` no?

albanD commented on torch/autograd/gradcheck.py:
```diff
@@ -191,13 +193,13 @@ def compute(a, b):
 
 
-def compute_numerical_jacobian_cols(jvp_fn, delta, input_is_complex) -> List[torch.Tensor]:
+def _compute_jvps_wrt_specific_input(jvp_fn, delta, input_is_complex) -> List[torch.Tensor]:
```
You want to keep the `numerical` vs `analytical` in these new namings.
Mostly because soon, we will add an `analytical` jvp function here.

albanD commented on torch/autograd/gradcheck.py:
```diff
@@ -291,21 +299,23 @@ def get_input_to_perturb(input):
 
 
-def get_jvp_fn(wrapped_fn, input_to_perturb, eps, nbhd_checks_fn):
+def _get_jvp_fn(wrapped_fn, input_to_perturb, eps, nbhd_checks_fn):
```
Here again, we're missing a numerical in the name I think.

albanD commented on torch/autograd/gradcheck.py:
```diff
@@ -848,9 +860,9 @@ def to_real_dtype(dtype):
-    # If x is complex, we create a complex tensor with only real component
+def _vec_from_tensor(x, generator, downcast_complex=False):
+    # Create a random vector with the same number of elements as x and the same
+    # dtype/device.
```
Why remove the comment about the special case for complex? You still do something special there.

soulitzer replied:
Whoops, this should still be there

soulitzer committed:
Update on "Clean up naming and comments"


This PR does many things but does not update any logic:
 - Prefixes all function names that are not `gradcheck`, `gradgradcheck`, `get_numerical_jacobian`, and `get_analytical_jacobian` with underscore to indicate that they aren't part of the public API (https://github.com/pytorch/pytorch/issues/55714).
 - Improve naming to avoid referencing Jacobian rows or Jacobian cols when we really mean vjp and jvp as suggested by @zou3519
 - Try to reduce comment line length so they are more consistent and easier to read
 - Other misc improvements to documentaiton


[ghstack-poisoned]

soulitzer committed:
Update on "Clean up naming and comments"


This PR does many things but does not update any logic:
 - Prefixes all function names that are not `gradcheck`, `gradgradcheck`, `get_numerical_jacobian`, and `get_analytical_jacobian` with underscore to indicate that they aren't part of the public API (https://github.com/pytorch/pytorch/issues/55714).
 - Improve naming to avoid referencing Jacobian rows or Jacobian cols when we really mean vjp and jvp as suggested by @zou3519
 - Try to reduce comment line length so they are more consistent and easier to read
 - Other misc improvements to documentaiton


[ghstack-poisoned]

albanD commented on torch/autograd/gradcheck.py:
```diff
@@ -112,16 +111,18 @@ def get_stride(size):
-    float/complex tensors.
+    """Computes the numerical Jacobian of `fn(inputs)` with respect to `target`. If
+    not specified, targets are the input. Returns M * N Jacobians where N is the
+    number of tensors in target that require grad and N is the number of non-integral
```
Should be M

albanD reviewed (approved):
One typo, otherwise LGTM

soulitzer committed:
Update on "Clean up naming and comments"


This PR does many things but does not update any logic:
 - Prefixes all function names that are not `gradcheck`, `gradgradcheck`, `get_numerical_jacobian`, and `get_analytical_jacobian` with underscore to indicate that they aren't part of the public API (https://github.com/pytorch/pytorch/issues/55714).
 - Improve naming to avoid referencing Jacobian rows or Jacobian cols when we really mean vjp and jvp as suggested by @zou3519
 - Try to reduce comment line length so they are more consistent and easier to read
 - Other misc improvements to documentaiton


[ghstack-poisoned]

soulitzer committed:
Update on "Clean up naming and comments"


This PR does many things but does not update any logic:
 - Prefixes all function names that are not `gradcheck`, `gradgradcheck`, `get_numerical_jacobian`, and `get_analytical_jacobian` with underscore to indicate that they aren't part of the public API (https://github.com/pytorch/pytorch/issues/55714).
 - Improve naming to avoid referencing Jacobian rows or Jacobian cols when we really mean vjp and jvp as suggested by @zou3519
 - Try to reduce comment line length so they are more consistent and easier to read
 - Other misc improvements to documentaiton

Differential Revision: [D28096571](https://our.internmc.facebook.com/intern/diff/D28096571)

[ghstack-poisoned]

soulitzer committed:
Update on "Clean up naming and comments"


This PR does many things but does not update any logic:
 - Prefixes all function names that are not `gradcheck`, `gradgradcheck`, `get_numerical_jacobian`, and `get_analytical_jacobian` with underscore to indicate that they aren't part of the public API (https://github.com/pytorch/pytorch/issues/55714).
 - Improve naming to avoid referencing Jacobian rows or Jacobian cols when we really mean vjp and jvp as suggested by @zou3519
 - Try to reduce comment line length so they are more consistent and easier to read
 - Other misc improvements to documentaiton

Differential Revision: [D28096571](https://our.internmc.facebook.com/intern/diff/D28096571)

[ghstack-poisoned]
